client/src/main/scala/client/BackPressuredWebSocketClient.scala

```diff
@@ -1,58 +1,67 @@
 package client
 
-import monifu.concurrent.Scheduler
-import monifu.reactive.{Ack, Observable, Observer, Subscriber}
+import monix.execution.cancelables.BooleanCancelable
+import monix.execution.{Ack, Cancelable, Scheduler}
+import monix.reactive.observers.Subscriber
+import monix.reactive.{Observable, Observer}
 import org.reactivestreams.Subscription
 import org.scalajs.dom.raw.MessageEvent
 import org.scalajs.dom.{CloseEvent, ErrorEvent, Event, WebSocket}
+
 import scala.concurrent._
-import concurrent.duration._
+import scala.concurrent.duration._
 
 
-final class BackPressuredWebSocketClient private (url: String)
-  extends Observable[String] { self =>
+final class BackPressuredWebSocketClient private(url: String)
+  extends Observable[String] {
+  self =>
 
   private def createChannel(webSocket: WebSocket) =
-    Observable.create[String] { subscriber =>
-      import subscriber.scheduler
-      val downstream = Observer.toReactiveSubscriber(subscriber)
-
-      try {
-        webSocket.onopen = (event: Event) => {
-          downstream.onSubscribe(new Subscription {
-            def cancel(): Unit =
-              closeConnection(webSocket)
-
-            def request(n: Long): Unit = {
-              webSocket.send(n.toString)
-            }
-          })
-        }
-
-        webSocket.onerror = (event: ErrorEvent) => {
-          downstream.onError(BackPressuredWebSocketClient.Exception(event.message))
-        }
-
-        webSocket.onclose = (event: CloseEvent) => {
-          downstream.onComplete()
+    new Observable[String] {
+      override def unsafeSubscribeFn(subscriber: Subscriber[String]): Cancelable = {
+        import subscriber.scheduler
+        val downstream = Observer.toReactiveSubscriber(subscriber)
+
+        try {
+          webSocket.onopen = (event: Event) => {
+            downstream.onSubscribe(new Subscription {
+              def cancel(): Unit =
+                closeConnection(webSocket)
+
+              def request(n: Long): Unit = {
+                webSocket.send(n.toString)
+              }
+            })
+          }
+
+          webSocket.onerror = (event: ErrorEvent) => {
+            downstream.onError(BackPressuredWebSocketClient.Exception(event.message))
+          }
+
+          webSocket.onclose = (event: CloseEvent) => {
+            downstream.onComplete()
+          }
+
+          webSocket.onmessage = (event: MessageEvent) => {
+            downstream.onNext(event.data.asInstanceOf[String])
+          }
+        } catch {
+          case ex: Throwable =>
```

Comment:
We shouldn't catch `Throwable` here. I think we should catch `NonFatal(ex)`. On top of Javascript it doesn't matter, but we shouldn't teach people bad practices.